allennlp/commands/predict.py

```diff
@@ -212,10 +221,21 @@ def _get_json_data(self) -> Iterator[JsonDict]:
                     yield self._predictor.load_line(line)
         else:
             input_file = cached_path(self._input_file)
-            with open(input_file, "r") as file_input:
-                for line in file_input:
-                    if not line.isspace():
-                        yield self._predictor.load_line(line)
+            try:
+                with open_compressed(input_file) as file_input:
+                    for line in file_input:
+                        if not line.isspace():
+                            yield self._predictor.load_line(line)
+            except OSError:
+                if self.compression_type:
+                    with open_compressed(input_file, self.compression_type) as file_input:
+                        for line in file_input:
+                            if not line.isspace():
+                                yield self._predictor.load_line(line)
+                else:
+                    print(
+                        "Automatic detection of compression type failed, please specify the compression type argument"
+                    )
```

Comment:
This logic needs to be the other way around. If the compression type is specified, we have to always respect it. If it's not specified, we autodetect.

Reply:
That makes sense, will incorporate it